node/src/actors/chain_manager/mod.rs

```diff
@@ -976,6 +976,27 @@ impl ChainManager {
                 }
             }
 
+            // Stop validating new commit transactions as soon as we achieve the limit of commits because
+            // the VRF validation is expensive. The limit is the number of witnesses of the data request.
+            if let Transaction::Commit(co_tx) = &msg.transaction {
+                let dr_pointer = co_tx.body.dr_pointer;
+                if let Some(dr_state) = self
+                    .chain_state
+                    .data_request_pool
+                    .data_request_state(&dr_pointer)
+                {
+                    let dr_output = &dr_state.data_request;
+                    let commit_limit = usize::from(dr_output.witnesses);
+                    let num_commits_in_pool = self
+                        .transactions_pool
+                        .num_commits_for_data_request(&dr_pointer);
+                    if num_commits_in_pool >= commit_limit {
```

Comment:
We can add some randomness here, for example we could validate 10% of the commits when over the limit. So if we have 1000 commits for a data request with 100 witnesses, this would result in a total of about 190 commits: the first 100 + 10% of the remaining 900.

```suggestion
                    if num_commits_in_pool >= commit_limit && rng.chance(0.1) {
```